pkg/initializers/dataset/cache.py

```diff
@@ -0,0 +1,295 @@
+                    "size": cluster_size,
+                    "leaderTemplate": {
+                        "metadata": {"labels": {"app": f"{train_job_name}-cache-head"}},
+                        "spec": {
+                            "serviceAccountName": f"{train_job_name}-sa",
+                            "containers": [
+                                {
+                                    "name": "head",
+                                    "image": cache_image,
+                                    "command": ["head"],
+                                    "args": ["0.0.0.0", "50051"],
+                                    "resources": {
+                                        "limits": {"cpu": head_cpu, "memory": head_mem},
+                                        "requests": {
+                                            "cpu": head_cpu,
+                                            "memory": head_mem,
+                                        },
+                                    },
+                                    "env": env_vars,
+                                    "ports": [{"containerPort": 50051}],
+                                }
+                            ],
+                        },
+                    },
+                    "workerTemplate": {
+                        "spec": {
+                            "serviceAccountName": f"{train_job_name}-sa",
+                            "containers": [
+                                {
+                                    "name": "worker",
+                                    "image": cache_image,
+                                    "command": ["worker"],
+                                    "args": ["0.0.0.0", "50051"],
+                                    "resources": {
+                                        "limits": {
+                                            "cpu": worker_cpu,
+                                            "memory": worker_mem,
+                                        },
+                                        "requests": {
+                                            "cpu": worker_cpu,
+                                            "memory": worker_mem,
+                                        },
+                                    },
+                                    "env": env_vars,
+                                    "ports": [{"containerPort": 50051}],
+                                }
+                            ],
+                        }
+                    },
+                },
+            },
+        }
+
+        # Create LeaderWorkerSet
+        custom_api.create_namespaced_custom_object(
+            group="leaderworkerset.x-k8s.io",
+            version="v1",
+            namespace=namespace,
+            plural="leaderworkersets",
+            body=lws_body,
+        )
+        logging.info(f"Created LeaderWorkerSet {train_job_name}-cache")
+
+        # Create Service
+        service = client.V1Service(
```

Comment:
LWS doesn't have any API to create service automatically ?

Maybe @kerthcet @kannon92 @ahg-g @ardaguclu  knows about it ?

Reply:
I think lws controller doesn't create/manage service objects and it is upto clients to define.